pkg/services/rpcsrv/client_test.go

```diff
@@ -2523,3 +2524,116 @@ func TestGetBlockNotifications(t *testing.T) {
 		require.NotEmpty(t, bn)
 	})
 }
+
+func TestClient_InvokeContainedScript(t *testing.T) {
+	chain, _, httpSrv := initServerWithInMemoryChain(t)
+
+	c, err := rpcclient.New(context.Background(), httpSrv.URL, rpcclient.Options{})
+	require.NoError(t, err)
+	t.Cleanup(c.Close)
+
+	var (
+		trig    = trigger.Application
+		verbose = true
+		tx      = &transaction.Transaction{
+			Version:         1,
+			Nonce:           2,
+			SystemFee:       3,
+			NetworkFee:      4,
+			ValidUntilBlock: 5,
+			Signers: []transaction.Signer{
+				{
+					Account: util.Uint160{1, 2, 3},
+					Scopes:  transaction.CalledByEntry,
+				},
+			},
+		}
+		h = &block.Header{
+			Version:       1,
+			PrevHash:      util.Uint256{1, 2, 3},
+			MerkleRoot:    util.Uint256{3, 2, 1},
+			Timestamp:     2,
+			Nonce:         3,
+			Index:         4,
+			NextConsensus: util.Uint160{4, 5, 6},
+			PrimaryIndex:  5,
+		}
+		txH    = util.Uint256{9, 8, 7}
+		txSize = 123
+		bH     = util.Uint256{6, 5, 4}
+	)
+	tx.SetHash(txH)
+	tx.SetSize(txSize)
+	h.SetHash(bH)
+
+	type testCase struct {
+		name     string
+		script   []byte
+		expected any
+	}
+	var (
+		testCases []testCase
+		src       = io.NewBufBinWriter()
+	)
+
+	emit.AppCall(src.BinWriter, nativehashes.LedgerContract, "currentIndex", callflag.All)
+	testCases = append(testCases, testCase{
+		name:     "Ledger.currentIndex",
+		script:   slices.Clone(src.Bytes()),
+		expected: h.Index - 1, // index of persisted block.
+	})
+
+	// TODO: @roman-khimov, pay attention that if we're prohibiting to use OnPersist/PostPersist triggers
+	// with `invokecontainedscript` then it's not possible to retrieve persisting block hash from the contract,
+	// hence hash mocking functionality is not needed for block.
```

Comment:
Hash is almost irrelevant, but height can be useful. Is it possible to override it properly? Otherwise we're left with time only.

Reply:
The current implementation allows to override height properly:

The only thing user has to know is that Ledger's `getCurrentHeight` returns *persisted* block height, not the *persisting* one.

Reply:
So if I'm to pass `block{Index: 1000}`, I'd get current height of 999? Works for us.